From e4824e0184738dd78bbdade26af6ee8d41db6a35 Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Wed, 7 Dec 2022 17:29:46 +0100
Subject: [PATCH] HMX DT: Setup T1 phys

Work in progress to get T1 phys communicating.
Trying with C45 generic driver.
PHYS responds to addr 3 should be 0.
---
 .../dts/freescale/imx8mp-var-dart-hmx1.dts    | 57 ++++++++++++-------
 1 file changed, 38 insertions(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
index 3de36d78f51e..496211fe2d74 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
@@ -242,32 +242,46 @@ can5_osc: can5_osc {
 	};
 };
 
-/* TODO: Adapt these when we have the phys designed in */
-/*
-&eqos {
-	mdio {
-		ethphy1: ethernet-phy@1 {
-			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <1>;
-			at803x,eee-disabled;
-			at803x,vddio-1p8v;
-			eee-broken-1000t;
-		};
-	};
-};
 
+/* First Automotive Ethernet Marvell 88Q2110 */
 &fec {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec>;
 	phy-mode = "rgmii";
-	phy-handle = <&ethphy1>;
-	phy-reset-gpios = <&pca6408_2 0 GPIO_ACTIVE_LOW>;
-	phy-reset-duration = <10>;
-	phy-reset-post-delay = <100>;
+	phy-handle = <&ethphy_t1_1>;
 	status = "okay";
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		ethphy_t1_1: ethernet-phy@0 {
+			// trying med clause 22 instead of clause 45
+			compatible = "ethernet-phy-ieee802.3-c45";
+			reg = <3>;
+			reset-gpios = <&gpio3 1 GPIO_ACTIVE_HIGH>;
+			at803x,eee-disabled;
+			eee-broken-1000t;
+			reset-assert-us = <10000>;
+			reset-deassert-us = <20000>;
+			vddio-supply = <&vddio1>;
+
+			vddio1: vddio-regulator {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+			};
+		};
+	};
 };
-*/
 
+/* Second Automotive Ethernet Marvell 88Q2110*/
+&eqos {
+	mdio {
+		ethernet-phy@0 {
+			// trying med clause 22 instead of clause 45
+			compatible = "ethernet-phy-ieee802.3-c45";
+			reg = <3>;
+		};
+	};
+};
 
 &i2c2 {
 	clock-frequency = <400000>;
@@ -637,8 +651,11 @@ MX8MP_IOMUXC_GPIO1_IO14__GPIO1_IO14	0x0
 		>;
 	};
 
+	/* 0x90 = Schmitt-trigger, fast slew-rate, 0x2 = Drive strength 4 (DSE_X4) */
 	pinctrl_fec: fecgrp {
 		fsl,pins = <
+			MX8MP_IOMUXC_SAI1_RXD2__ENET1_MDC				0x2
+			MX8MP_IOMUXC_SAI1_RXD3__ENET1_MDIO				0x2
 			MX8MP_IOMUXC_SAI1_RXD4__ENET1_RGMII_RD0				0x90
 			MX8MP_IOMUXC_SAI1_RXD5__ENET1_RGMII_RD1				0x90
 			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x90
@@ -651,6 +668,8 @@ MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x00
 			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x00
 			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x00
 			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x00
+			MX8MP_IOMUXC_NAND_CE0_B__GPIO3_IO01				0x02
+
 		>;
 	};
 
@@ -930,7 +949,7 @@ &gpio2 {
 
 &gpio3 {
 	gpio-line-names = "USB_UPGRADE_BUTTON",
-		"USER_LED_2_PWR",
+		"A_ETH_nRST",
 		"GPIO3_IO02",
 		"GPIO3_IO03",
 		"GPIO3_IO04",
-- 
2.30.2

